qiling/os/memory.py

```diff
@@ -268,12 +280,12 @@ def save(self):
             "mmio" : []
         }
 
-        for lbound, ubound, perm, label, is_mmio in self.map_info:
+        for lbound, ubound, perm, label, is_mmio, data in self.map_info:
             if is_mmio:
                 mem_dict['mmio'].append((lbound, ubound, perm, label, *self.mmio_cbs[(lbound, ubound)]))
             else:
-                data = self.read(lbound, ubound - lbound)
-                mem_dict['ram'].append((lbound, ubound, perm, label, bytes(data)))
+                data = self.read(lbound, ubound - lbound)  # read instead of using data from map_info to avoid error
+                mem_dict['ram'].append((lbound, ubound, perm, label, data))
```

Comment:
Actually data in `map_info` will be different from the return of `uc.read`, which causes error in ql.mem.save/restore, so I made a hack here. But it seems to cover some unseen inconsistency between `map_info` and `uc.mem`, which could be source of other errorrs.